### PR TITLE
Enhance agent association editing

### PR DESCRIPTION
**Pull request by knu**
Here's what this PR is about:

- Receivers are inherited when cloning an agent.
- Controllers are selectable from the agent edit form.

### CODE REVIEW

File: app/views/agents/_form.html.erb [hunk 1]
```diff
@@ -51,13 +51,14 @@
               </div>
             </div>
 
-            <div class="controller-region" data-has-controllers="<%= !@agent.controllers.empty? %>">
+            <div class="controller-region">
```
Comment (dsander) [inline app/views/agents/_form.html.erb]: Can you also remove the associated stylesheets?

File: Gemfile [hunk 2]
```diff
@@ -149,6 +149,7 @@ group :development do
   group :test do
     gem 'coveralls', '~> 0.7.4', require: false
     gem 'capybara', '~> 2.13.0'
+    gem 'capybara-screenshot'
```
Comment (dsander) [inline Gemfile]: Does this gem do more than `save_and_open_screenshot` offers?

Comment (knu) [inline Gemfile]: It automatically saves screenshots in html and png for each feature spec failure.  You'll be free of having to manually do that and rerun the failed specs.

File: app/views/agents/_form.html.erb (hunk 1 above)
Comment (knu) [inline app/views/agents/_form.html.erb]: Done!

**Comment by cantino** [comment]
Nice :)

**Comment by knu** [comment]
@dsander @cantino Thanks for taking a look!

**Pull request merged**